### ISSUE TITLE
Settle on backends for jobs and data

### ISSUE THREAD

**Issue by dometto** [repotag/cutting_edge#8]
## Jobs

Possibilities:

* Sidekiq (backend: Redis)
* DelayedJob (backend: SQL)
  * Looks a bit Rails-centric, but perhaps not impossible to get work [with sinatra](https://gist.github.com/robhurring/732327)
* Resque (backend: Redis)
  * No multithread support, so better to use Sidekiq.

There is also [rufus-scheduler](https://github.com/jmettraux/rufus-scheduler), which is multithreaded and in-memory (without a backend). It doesn't look like a full-fledged async 'Worker' solution like the above three, but maybe it's good enough for us?

## Data

What data:

* Which repositories to monitor?
  * currently YAML
* Stored/cached badges
* Stored/cached results on dependencies

Options:

* SQLlite
* Redis
* On-disk
  * Downside: wouldn't work on Heroku (due to their ephemeral file system)

If we go for Redis for both jobs and data, we should perhaps [use different databases](https://www.mikeperham.com/2015/09/24/storing-data-with-redis/) for both.

**Comment by dometto** [comment]
I was thinking this might be a cool app to deploy to Heroku using its free plan, but it appears there are some issues with that: https://stackoverflow.com/questions/22997650/rufus-scheduler-and-delayed-job-on-heroku-why-use-a-worker-dyno

TL;DR apps on Heroku's free plan idle after an hour, so scheduled tasks wouldn't run.

**Comment by dometto** [comment]
We've settled on two better options:
* [SuckerPunch](https://github.com/brandonhilkert/sucker_punch) for workers
  * Runs in the same process as the app
  * No need for job persistence
  * Use [whenever](https://github.com/javan/whenever) to schedule recurring jobs
* [Moneta](https://github.com/moneta-rb/moneta) key/value store for data
  * Backend agnostic, so can use in-memory, redis, etc.

**Comment by dometto** [comment]
Resolved in #9
